devito/types/basic.py

```diff
@@ -1352,13 +1383,14 @@ def indexify(self, indices=None, subs=None):
 
         # Indices after substitutions
         indices = []
-        for a, d, o, s in zip(self.args, self.dimensions, self.origin, subs):
+        for a, d, o, of, s in zip(self.args, self.dimensions, self.origin,
+                                  self._offset_subdomain, subs):
             if d in a.free_symbols:
-                # Shift by origin d -> d - o.
-                indices.append(sympy.sympify(a.subs(d, d - o).xreplace(s)))
+                # Shift by origin and offset d -> d - o - of.
```

Comment:
Would it make sense to have origin just directly return the right one, or maybe `local_origin` ?